sdk/lib/opentelemetry/sdk/internal.rb

```diff
@@ -16,14 +16,39 @@ def boolean?(value)
         value.is_a?(TrueClass) || value.is_a?(FalseClass)
       end
 
+      def homogenous_element_types?(value)
+        first_element_type = nil
+        value.each do |element|
+          element_type = case element
+                         when String then String
+                         when TrueClass, FalseClass then :boolean
+                         when Numeric then Numeric
+                         end
+          if !first_element_type
+            first_element_type = element_type
+          elsif element_type != first_element_type
+            return false
+          end
+        end
+        true
+      end
+
```

Comment:
It might be a improvement if we remove this, and then write things as I have in the following comment.